### ISSUE TITLE
[Spree Upgrade] Extract #tax_adjustment_totals and test it properly

### ISSUE THREAD

**Issue by sauloperez** [openfoodfoundation/openfoodnetwork#3495]
### What we should change and why (this is tech debt)

The following method is way to big to be in the already bloated order decorator

https://github.com/openfoodfoundation/openfoodnetwork/blob/0a885af7a6165bf19809304fa0232592c646300d/app/models/spree/order_decorator.rb#L294-L303

What's worse is that the lack of encapsulation (and so little maneuverability to improve it) led to dubious and very hard to follow tests

https://github.com/openfoodfoundation/openfoodnetwork/blob/0a885af7a6165bf19809304fa0232592c646300d/spec/models/spree/order_spec.rb#L281-L330

### Context

This came up in PR #3299. It took me a bit to wrap my head around it and I wasn't very confident of its test coverage.

### Impact and timeline

This would potentially increase its test coverage while also making the method more maintainable. Extracting it out of the order decorator would also make that class a favor.

**Comment by luisramos0** [comment]
Removing [Spree Upgrade] so we keep the tag for issues that are mandatory for v2.
